### PR TITLE
Restrict station IDs to letters & numbers

### LINKED ISSUE #68: Restrict station IDs to letters & numbers

### PR DESCRIPTION
**Pull request by t0stiman**
closes #68

### CODE REVIEW

**Comment by Insprill** [comment]
It'd be a lot simpler to match it against a regex pattern like `^[a-zA-Z0-9]*$` and just have the message say it contains invalid characters.

**Comment by t0stiman** [comment]
@Insprill like this?

File: MapifyEditor/Export/Validators/GameContent/StationValidator.cs [hunk 1]
```diff
@@ -26,8 +27,16 @@ protected override IEnumerator<Result> Validate(Scenes scenes)
 
                 if (string.IsNullOrWhiteSpace(station.stationName))
                     yield return Result.Error($"Station '{station.name}' must have a name", station);
+
                 if (string.IsNullOrWhiteSpace(station.stationID))
                     yield return Result.Error($"Station '{station.name}' must have an ID", station);
+
+                var alphaNumericRegex = new Regex(@"^[a-zA-Z0-9]*$");
```
Comment (Insprill) [inline MapifyEditor/Export/Validators/GameContent/StationValidator.cs:34]: You should be able to use the static Regex.IsMatch method here. If not we should statically cache it in the class.

**Review by Insprill — CHANGES_REQUESTED** [review]

File: MapifyEditor/Export/Validators/GameContent/StationValidator.cs (hunk 1 above)
Comment (t0stiman) [reply MapifyEditor/Export/Validators/GameContent/StationValidator.cs]: changed

**Pull request merged**